apps/ui/src/Communities/CommunitiesFeed.tsx

```diff
@@ -1,45 +1,83 @@
 import Box from '@mui/material/Box';
 import CommunityCard from './CommunityCard/CommunityCard';
-import { IconButton, Tooltip } from '@mui/material';
+import { IconButton } from '@mui/material';
 import AddIcon from '@mui/icons-material/Add';
 
 import defaultTheme from '../themes/default';
 import { Community } from '@communecar/types';
 import { useUserCommunitiesStatus } from '../hooks/Communities/useUserCommunitiesStatus';
+import { SearchBar } from '../Components/Search/SearchBar';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { HeaderFeed } from './HeaderFeed.styled';
+import { ItemsListFeed } from './ItemsListFeed.styled';
 
 export interface CommunitiesFeedProps {
   communities: Community[];
 }
 
 const CommunitiesFeed = ({ communities }: CommunitiesFeedProps) => {
   const userCommunitiesStatus = useUserCommunitiesStatus('hi');
+  const [filteredCommuniuties, setFilteredCommuniuties] = useState(communities);
+  const [searchValue, setSearchValue] = useState<string>();
+
+  const options = useMemo(
+    () => communities.map((community) => community.name),
+    [communities],
+  );
+
+  const handleChangeSearchValue = (value: string | undefined) => {
+    const lowerCaseValue = value?.toLowerCase();
+    setSearchValue(lowerCaseValue);
+    filterDisplay(lowerCaseValue);
+  };
+
+  const filterDisplay = useCallback(
+    (value: string | undefined) => {
+      const newFilteredCommuniuties = value
+        ? communities.filter((community) =>
+            community.name.toLowerCase().includes(value),
+          )
+        : communities;
+      setFilteredCommuniuties(newFilteredCommuniuties);
+    },
+    [communities],
+  );
+
+  useEffect(() => {
```

Comment:
Why do you need this useEffect?

Reply:
If a new community is added while a user is looking at the communities in a filtered way then it checks the filter again and accordingly puts the new community in its view or doesn't

Reply:
You need to send a request to the server to get the communities, so a new community won't magically appear in the list.
We can add a filter route to the backend that gets the search value and returns communities according to it.

Reply:
Reminder

Reply:
This is something I talked to with Tal - when creating a new community we put the newly created community into the list in the UI and at the same time update the BE and then when the list of communities is updated it will update here. At the moment I only added the functionality that the UI does